src/commands/search.ts

```diff
@@ -1,19 +1,41 @@
 import { GluegunCommand, GluegunToolbox } from 'gluegun';
-import { mail } from '../lib/mail';
+import mail from '../hooks/mail';
+import pushover from '../hooks/pushover';
+import { DiningAvailability } from '../disney-api/model/response';
 
 module.exports = {
   name: 'search',
   run: async (toolbox: GluegunToolbox) => {
-    const { parameters, print, disneyApi } = toolbox;
-    const date = parameters.first;
+    const {
+      parameters: { options },
+      print,
+      disneyApi,
+    } = toolbox;
 
-    const onSuccess = (foundTables) => {
-      console.log('Found a table!');
-      console.log(foundTables);
-      mail('Be our guest!', JSON.stringify(foundTables));
-    };
+    const defaultDate = new Date();
+    const year = new Intl.DateTimeFormat('en', { year: 'numeric' }).format(defaultDate);
+    const month = new Intl.DateTimeFormat('en', { month: 'numeric' }).format(defaultDate);
+    const day = new Intl.DateTimeFormat('en', { day: '2-digit' }).format(defaultDate);
 
-    print.info(`Checking for tables on ${date}...`);
-    disneyApi.checkTables(date, onSuccess);
+    const { date = `${year}-${month}-${day}`, ids, party = 2 } = options;
+
+    //Hooks
+    const onSuccess = async ({ diningAvailability }: { diningAvailability: DiningAvailability }) =>
+      Promise.allSettled([
+        mail({ diningAvailability, print, partySize: party, date }),
+        pushover({ diningAvailability, print, partySize: party, date }),
+      ]);
+
+    if (ids) {
+      disneyApi.checkTables({
+        date,
+        onSuccess,
+        print,
+        ids: (ids && typeof ids === 'number' ? ids.toString() : ids).split(','),
```

Comment:
unnecessary to check `ids` existence again in this if block

```suggestion
        ids: (typeof ids === 'number' ? ids.toString() : ids).split(','),
```

Reply:
Oh right. I'm surprised typescript didn't complain at me about that.